### ISSUE TITLE
[Stdlib] Add List.find and List.findIndex methods

### ISSUE THREAD

**Issue by ospencer** [grain-lang/grain#348]
Here's the type for `find`:

```grain
find : ((a -> Bool), List<a>) -> a
```

It should take a predicate and a list, and return the first item that matches the predicate. It should fail if no item matches the predicate.

And here's the type for `findIndex`:

```grain
findIndex : ((a -> Bool), List<a>) -> Number
```

It's the same as `find`, but returns the position of the first matching item (from 0). Also fails if no item matches the predicate.

**Comment by ohana54** [comment]
I'll give this a go :)

**Comment by ohana54** [comment]
I couldn't find how to catch exceptions (need it to test the failure cases), could you please help me with that?

**Comment by ospencer** [comment]
Wasm languages still have trouble handling exceptions, unfortunately: https://github.com/WebAssembly/exception-handling

We've thought a bit about a custom implementation, but I think the current plan is to wait for official wasm support. We've talked about just not having these throw errors, though: https://github.com/grain-lang/grain/issues/159

**Comment by ospencer** [comment]
But the tl;dr is don't worry about testing it just yet :)

**Comment by ospencer** [comment]
Fixed in #353
